### PR TITLE
Don't queue up log messages when the process isn't running a Quarkus application

### LINKED ISSUE #14295: java.lang.OutOfMemoryError: Java heap space - Quarkus 1.10.5.Final
**Describe the bug**
Difference in memory consumption between version 1.8.3 and 1.10.* 
It can result in an OutOfMemoryError on a simple unit test using the ApachDS library.

**Expected behavior**
Use of 300 MB of memory instead of 2 GB +
Build succes with the unit test. 

**Actual behavior**
```java
Running org.nch.GreetingResourceTest
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 42.689 sec <<< FAILURE!
org.nch.GreetingResourceTest  Time elapsed: 42.689 sec  <<< ERROR!
java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:3332)
	at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:124)
	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:448)
	at java.lang.StringBuilder.append(StringBuilder.java:136)
	at java.lang.StringBuilder.append(StringBuilder.java:131)
	at org.apache.directory.api.ldap.model.schema.registries.Registries.dumpUsing(Registries.java:1954)
	at org.apache.directory.api.ldap.model.schema.registries.Registries.addReference(Registries.java:2031)
	at org.apache.directory.api.ldap.model.schema.registries.helper.ObjectClassHelper.addToRegistries(ObjectClassHelper.java:81)
	at org.apache.directory.api.ldap.model.schema.registries.Registries.buildReference(Registries.java:690)
	at org.apache.directory.api.ldap.model.schema.registries.Registries.buildObjectClassReferences(Registries.java:815)
	at org.apache.directory.api.ldap.model.schema.registries.Registries.buildReferences(Registries.java:875)
	at org.apache.directory.api.ldap.model.schema.registries.Registries.clone(Registries.java:2578)
	at org.apache.directory.api.ldap.schemamanager.impl.DefaultSchemaManager.cloneRegistries(DefaultSchemaManager.java:200)
	at org.apache.directory.api.ldap.schemamanager.impl.DefaultSchemaManager.load(DefaultSchemaManager.java:589)
	at org.apache.directory.server.core.api.schema.registries.synchronizers.SchemaSynchronizer.add(SchemaSynchronizer.java:215)
	at org.apache.directory.server.core.api.schema.registries.synchronizers.RegistrySynchronizerAdaptor.add(RegistrySynchronizerAdaptor.java:188)
	at org.apache.directory.server.core.api.schema.SchemaPartition.add(SchemaPartition.java:266)
	at org.apache.directory.server.core.shared.partition.DefaultPartitionNexus.add(DefaultPartitionNexus.java:385)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor$1.add(BaseInterceptor.java:158)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)
	at org.apache.directory.server.core.journal.JournalInterceptor.add(JournalInterceptor.java:139)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)
	at org.apache.directory.server.core.changelog.ChangeLogInterceptor.add(ChangeLogInterceptor.java:113)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)
	at org.apache.directory.server.core.trigger.TriggerInterceptor.add(TriggerInterceptor.java:300)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)
	at org.apache.directory.server.core.event.EventInterceptor.add(EventInterceptor.java:220)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)
	at org.apache.directory.server.core.subtree.SubentryInterceptor.add(SubentryInterceptor.java:1014)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)
	at org.apache.directory.server.core.collective.CollectiveAttributeInterceptor.add(CollectiveAttributeInterceptor.java:134)
	at org.apache.directory.server.core.api.interceptor.BaseInterceptor.next(BaseInterceptor.java:412)

Results :

Tests in error: 
  org.nch.GreetingResourceTest: Java heap space
```

**To Reproduce**

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/5813489/code-with-quarkus.zip)

1. mvn clean install 

**Environment (please complete the following information):**
 - #30-Ubuntu SMP Mon Apr 20 16:58:30 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
 - openjdk version "1.8.0_232"
   OpenJDK Runtime Environment (build 1.8.0_232-8u232-b09-0ubuntu1~19.04.1-b09)
   OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)
 - Quarkus version or git rev :  1.10.5.Final
 - Apache Maven 3.6.3 
   Maven home: /opt/apache-maven-3.6.3
   Java version: 1.8.0_232, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
   Default locale: fr_FR, platform encoding: UTF-8
   OS name: "linux", version: "5.4.0-26-generic", arch: "amd64", family: "unix"

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #14295

### CODE REVIEW

File: independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java [hunk 1]
```diff
@@ -14,25 +14,41 @@
...
+                if (root.getClassLoader() != cl) {
+                    delayedHandler = (DelayedHandler) root.getDeclaredField("DELAYED_HANDLER").get(null);
+                }
+            } catch (Exception e) {
+                //ignore, happens on native image build
             }
-        } catch (Exception e) {
-            //ignore, happens on native image build
+        } catch (ClassNotFoundException e) {
+            isQuarkusApplication = false;
```
Comment (gsmet) [inline independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java:42]: Shouldn't we have a better way to know that?

Comment (geoand) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java:42]: I couldn't think of any given the fact that this class is loaded very very early...

**Comment by geoand** [comment]
This approach doesn't seem to work - if falls over during testing

**Comment by geoand** [comment]
I added a different approach

**Pull request closed without merging**